### PR TITLE
feat: Add module for AWS VPC endpoints

### PR DESCRIPTION
**Pull request by lzianekhodja**
Add a module to create a list of AWS VPC endpoints by giving in the input a map of information about each VPC endpoint.

### CODE REVIEW

**Review by YKharouni — APPROVED** [review]

File: networking/aws/vpce/examples/complete/main.tf [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    app             = "complete"
+    module          = "AWS VPCE"
+    "create by"     = data.aws_caller_identity.current.arn
+    "creation date" = null_resource.timestamp.triggers["creation_date"]
+  }
+}
+
+# Create a VPC endpoint for S3
+module "vpce" {
+  source = "../.."
```
Comment (lemaitre-aneo) [inline networking/aws/vpce/examples/complete/main.tf:53]:
```suggestion
  source = "../../../vpce"
```

File: networking/aws/vpce/examples/simple/main.tf [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    app             = "simple"
+    module          = "AWS VPCE"
+    "create by"     = data.aws_caller_identity.current.arn
+    "creation date" = null_resource.timestamp.triggers["creation_date"]
+  }
+}
+
+# Create a VPC endpoint for S3
+module "vpce" {
+  source = "../.."
```
Comment (lemaitre-aneo) [inline networking/aws/vpce/examples/simple/main.tf:52]:
```suggestion
  source = "../../../vpce"
```

File: networking/aws/vpce/variables.tf [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+variable "security_group_ids" {
+  description = "The IDs of security groups to associate with the network interfaces"
+  type        = list(string)
+  default     = []
+}
+
+variable "subnet_ids" {
+  description = "The IDs of subnets in which to create the network interfaces for the endpoints"
+  type        = list(string)
```
Comment (lemaitre-aneo) [inline networking/aws/vpce/variables.tf:14]:
```suggestion
  type        = set(string)
```

File: networking/aws/vpce/variables.tf [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  default     = []
+}
+
+variable "endpoints" {
+  description = "A map of interface and/or gateway endpoints containing their properties and configurations"
+  type        = any
```
Comment (lemaitre-aneo) [inline networking/aws/vpce/variables.tf:20]:
```suggestion
  type        = map(any)
```

**Review by lemaitre-aneo — COMMENTED** [review]
I wonder if we could use optionals in parameters. This would make the code much simpler. Would it be reasonable to make a hard dependency on terraform 1.3?

File: networking/aws/vpce/examples/complete/main.tf (hunk 1 above)
Comment (lzianekhodja) [reply networking/aws/vpce/examples/complete/main.tf:53]: done

File: networking/aws/vpce/examples/simple/main.tf (hunk 2 above)
Comment (lzianekhodja) [reply networking/aws/vpce/examples/simple/main.tf:52]: Done

File: networking/aws/vpce/variables.tf (hunk 4 above)
Comment (lzianekhodja) [reply networking/aws/vpce/variables.tf:20]: Done

**Comment by lzianekhodja** [comment]
I put in the file versions.tf that the minimum version for Terraform is 1 (required_version = ">= 1.0")

**Review by YKharouni — APPROVED** [review]

**Pull request merged**